### PR TITLE
Update the dockerfile for alpine image

### PR DESCRIPTION
**Pull request by hustcer**
# Description

Update the dockerfile for alpine image, related https://github.com/nushell/nushell/issues/14171 :

1.  Add armv7 arch support
2. Add more opencontainers labels, see: https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys

# User-Facing Changes

- Nushell docker images will be provided for each official release
- There will be a `nushell:nightly` tag for the latest nightly release

# After Submitting

1. The workflow that build the images will be provided in [nushell/nightly](https://github.com/nushell/nightly) repo later
2. Alpine image will be provided as the default, we may also add Debian images later

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
-LABEL maintainer=nushell
-
 ARG TARGETARCH
 ARG ARCH=${TARGETARCH/arm64/aarch64}
+ARG ARCH=${ARCH/arm/armv7}
 ARG ARCH=${ARCH/amd64/x86_64}
 
+ARG BUILD_REF
+ARG BUILD_DATE
+LABEL maintainer="Nushell Team" \
```
Comment (fdncred) [inline docker/Dockerfile:17]: Seems like some of this stuff should match our cargo.toml file.
https://github.com/nushell/nushell/blob/2c31b3db072f60869dbe49f3bb1943d76c07b21e/Cargo.toml#L2-L14

We don't use the term "Nushell Team" specifically.

**Comment by vyadh** [comment]
FYI, This morning created a Debian/Ubuntu Dockerfile based on this. I'll follow the changes here and raise a PR when this is ready.

File: docker/Dockerfile (hunk 1 above)
Comment (hustcer) [reply docker/Dockerfile]: I have just updated

**Comment by fdncred** [comment]
Thanks

**Pull request merged**